File: backend/src/main/java/com/app/mevocab/internal/vocabulary/entity/Word.java
package com.app.mevocab.internal.vocabulary.entity;

import com.app.mevocab.internal.topic.entity.Topic;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document
@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Word {
    @Id
    String word;
    String translation;
    Topic topic;
    String phonetic;
    String audio;
    List<String> partOfSpeeches;
}

File: frontend/lib/data/models/Word.dart
import 'Topic.dart';

class Word {
  final String word;
  final String translation;
  final Topic? topic;
  final String phonetic;
  final String audio;
  final List<String> partOfSpeeches;

  Word({
    required this.word,
    required this.translation,
    this.topic,
    required this.phonetic,
    required this.audio,
    required this.partOfSpeeches,
  });

  factory Word.fromJson(Map<String, dynamic> json) {
    return Word(
      word: json['word'] ?? '',
      translation: json['translation'] ?? '',
      topic: json['topic'] != null
          ? Topic.fromJson(json['topic'] as Map<String, dynamic>)
          : null,
      phonetic: json['phonetic'] ?? '',
      audio: json['audio'] ?? '',
      partOfSpeeches: (json['partOfSpeeches'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'word': word,
      'translation': translation,
      'topic': topic?.toJson(),
      'phonetic': phonetic,
      'audio': audio,
      'partOfSpeeches': partOfSpeeches,
    };
  }
}
